### DISCUSSION TITLE
Antizapret Heals Discord

### DISCUSSION THREAD

**Discussion by Selys-u** [xtrime-ru/antizapret-vpn-docker#96] [Q&A] [answered]
Привет. Получится вернуть дискорду жизнь?

**Comment by MascotWorld** [comment]
https://gitlab.com/GhostRooter0953/discord-voice-ips добавил отсюда ip + domain, не помогло

**Reply by Selys-u** [reply]
Я пробовал с другого ресурса список сетей через вг пустить, но войс не работает

**Comment by Ernieleo** [comment]
Создал регулярное выражение для фильтрации известных доменов Discord. 
Добавьте его в файл **config/custom/include-regex-custom.txt**. 

С ним работают сообщения, медиа-файлы, подгружаются мини-игры и работает магазин, обновляется и загружается Discord. Однако соединение с RTC не устанавливается, поскольку оно осуществляется по IP, а текущий контейнер с АнтиЗапретом со списком IP-адресов не функционирует. Похоже, что необходима дополнительная настройка для корректной работы с IP.
Жетально чтобы подключало к серверам в России или Финляндии... 

### Используемое регулярное выражение:
`^([a-zA-Z0-9-]+\.)*(discord|dis)(-attachments-uploads-prd\.storage|)(-activities|app|cdn|design|dev|gg|gift|gifts|media|new|store|tools|activities|partygames|says|sez|status)?\.[a-zA-Z0-9-]{2,}(\.com|\.net|\.co|\.gd|\.googleapis\.com)?$`

### Список поддерживаемых доменов:

discord.com
gateway.discord.gg
cdn.discordapp.com
discordapp.net
discord-attachments-uploads-prd.storage.googleapis.com
dis.gd
discord.co
discord.design
discord.dev
discord.gg
discord.gift
discord.gifts
discord.media
discord.new
discord.store
discord.tools
discordapp.com
discordmerch.com
discordpartygames.com
discord-activities.com
discordactivities.com
discordsays.com
discordstatus.com
googleapis.com
discordcdn.com
discordsez.com

upd. Исправил опечатку в какой файл добавлять.

**Reply by Selys-u** [reply]
Спасибо вам, мил человек. Огромное. Если бы были у нас ip, можно было бы статикой через туннель пускать

**Reply by Murakumo-JP** [reply]
Я либо чёт делаю не так либо я тупой, после добавления регулярки в файл include-hosts-custom у меня перестаёт работать амнезияWG

**Reply by Selys-u** [reply]
> Я либо чёт делаю не так либо я тупой, после добавления регулярки в файл include-hosts-custom у меня перестаёт работать амнезияWG

Там надо в файл с регулярным выражениями - include-regex-custom.txt

**Reply by Murakumo-JP** [reply]
Тут пишут что войсы тоже работают с этим списком и ip, может как то можно и это в регулярку впихнуть
https://ntc.party/t/discord-%D1%81%D0%BF%D0%B8%D1%81%D0%BE%D0%BA-%D0%B7%D0%B0%D0%B1%D0%BB%D0%BE%D0%BA%D0%B8%D1%80%D0%BE%D0%B2%D0%B0%D0%BD%D0%BD%D1%8B%D1%85-%D0%B4%D0%BE%D0%BC%D0%B5%D0%BD%D0%BE%D0%B2-%D0%B8-%D0%B0%D0%B4%D1%80%D0%B5%D1%81%D0%BE%D0%B2-%D0%B4%D0%BB%D1%8F-nekoray/11247

**Reply by Selys-u** [reply]
> Тут пишут что войсы тоже работают с этим списком и ip, может как то можно и это в регулярку впихнуть
> https://ntc.party/t/discord-%D1%81%D0%BF%D0%B8%D1%81%D0%BE%D0%BA-%D0%B7%D0%B0%D0%B1%D0%BB%D0%BE%D0%BA%D0%B8%D1%80%D0%BE%D0%B2%D0%B0%D0%BD%D0%BD%D1%8B%D1%85-%D0%B4%D0%BE%D0%BC%D0%B5%D0%BD%D0%BE%D0%B2-%D0%B8-%D0%B0%D0%B4%D1%80%D0%B5%D1%81%D0%BE%D0%B2-%D0%B4%D0%BB%D1%8F-nekoray/11247

Домены в файл hosts по идее можно добавить, а айпи отдельно через туннель, если есть. Предполагаю

**Reply by Murakumo-JP** [reply]
попытался добавить всё в файл host и впн тупа сдох он проста перестал грузить, убираешь всё с файла host и снова всё работает, бред

**Reply by Ernieleo** [reply]
> попытался добавить всё в файл host и впн тупа сдох он проста перестал грузить, убираешь всё с файла host и снова всё работает, бред

Как редактируете? Через консоль или через файл?
Рекомендую (из-за некоторых случаев) добавлять через консоль, если мою регулярку, то в include-regex-custom.txt, опечатлся ранее когда писал сообщение, исправил.

**Reply by Murakumo-JP** [reply]
> > попытался добавить всё в файл host и впн тупа сдох он проста перестал грузить, убираешь всё с файла host и снова всё работает, бред
> 
> Как редактируете? Через консоль или через файл? Рекомендую (из-за некоторых случаев) добавлять через консоль, если мою регулярку, то в include-regex-custom.txt, опечатлся ранее когда писал сообщение, исправил.

добавлял хосты через файл но впн помирал, регулярку добавил через файл вроде работает всё ок кроме войса как раз, а как через консоль это сделать ? я проста в этом вообще не селён ибо с убунту никогда не работал.

**Reply by Ernieleo** [reply]
@Murakumo-JP

Чтобы подключиться к контейнеру, используйте команду в ssh: **docker exec -it antizapret-vpn bash**
После этого:
- Перемещайтесь по директориям с помощью команды cd.
- Просматривайте файлы с помощью ls.
- Для редактирования файлов используйте: **nano имя_файла**

**Reply by Murakumo-JP** [reply]
> @Murakumo-JP
> 
> Чтобы подключиться к контейнеру, используйте команду в ssh: **docker exec -it antizapret-vpn bash** После этого:
> 
> * Перемещайтесь по директориям с помощью команды cd.
> * Просматривайте файлы с помощью ls.
> * Для редактирования файлов используйте: **nano имя_файла**

Спасибо завтра наверное уже попробую а то сегодня весь вечер убил чтоб поднять и настроить амнезиюWG а то до этого был опенвпн но провайдер закрыл доступ к нему теперь.

**Comment by xtrime-ru** [comment, marked as answer]
Я бы рекомендовал попробовать собрать список подсетей самостоятельно. Например так: 

1. Пускаем весь траф через wireguard. Для этого в клиентском конфиге прописываем AllowedIps = 0.0.0.0/0
2. В контейнере wireguard устанавливем `apk add tcpdump`
3. Запустить логирование UPD пакетов примерно так: `tcpdump -i any -n ip | grep UDP | grep -v 10.224`
4. Позвонить через discord и посмотреть куда идут UDP пакеты.

**Reply by Ernieleo** [reply]
Считаю, что на данный момент нет смысла углубляться в это, так как видел сообщения о том, что люди начали парсить все маршруты через API и готовят обновленные списки. На текущий момент добавленные в обновлении маршруты работают — судя по тестам, проведенным мной и друзьями в Европейской части России.

Решение, основанное на моей регулярке, успешно функционирует. Оно обеспечивает полный доступ, включая ранее отключенный в России Discord'ом сервис магазина с украшениями профиля, где доступны, в том числе, и бесплатные элементы.

На данный момент Discord признает проблему блокировок в России и Турции и, судя по статусу на [странице](https://discordstatus.com/), работает над ее решением. Вероятно, при обоюдном желании сторон они смогут договориться о разблокировке.

**Comment by Murakumo-JP** [comment]
Как я понял войс так и не вышла починить на AmneziaWG ? 😕, я пытался маршруты добавлять в свой docker-compose.override.yml но чёт не пашет нечего.

**Reply by Selys-u** [reply]
Я обновил антизапрет. Настроил через опенвпн и все заработало) войс в том числе.

Думаю, вам стоит обновиться. Технически, думаю, разницы нет, амнезия или опенвпн

**Reply by Murakumo-JP** [reply]
> Я обновил антизапрет. Настроил через опенвпн и все заработало) войс в том числе.
> 
> Думаю, вам стоит обновится

У меня проста настроена на Кинетике всё, опенвнп блочится провайдером к сожалению, в конфиг файле пускается весь трафик типа WG_ALLOWED_IPS=0.0.0.0/0 хотя на роутере он пускает не весь туда трафик, я пытался пихать маршруты дискорда но почему то дискорд трафик UDP не хочет работать, настраивал так https://github.com/xtrime-ru/antizapret-vpn-docker/blob/master/docs/guide_Keenetic_RU.md#wireguard 🤔

**Reply by Selys-u** [reply]
Настрою обратно через амнезию, сообщу заработает или нет. Только пока не могу сказать когда((
